orchestrator/schedules/service.py

```diff
@@ -0,0 +1,279 @@
+
+    Args:
+        payload: APSchedulerJobCreate | APSchedulerJobUpdate | APSchedulerJobDelete The scheduled task payload.
+    """
+    data = json.loads(payload.model_dump_json())
+    data["scheduled_type"] = payload._scheduled_type
+    return json.dumps(data).encode()
+
+
+def deserialize_payload(bytes_dump: bytes) -> APSchedulerJobCreate | APSchedulerJobUpdate | APSchedulerJobDelete:
+    """Deserialize the payload from bytes for Redis retrieval.
+
+    Args:
+        bytes_dump: bytes The serialized payload.
+    """
+    json_dump = bytes_dump.decode()
+
+    if SCHEDULER_Q_UPDATE in json_dump:
+        return APSchedulerJobUpdate.model_validate_json(json_dump)
+    if SCHEDULER_Q_DELETE in json_dump:
+        return APSchedulerJobDelete.model_validate_json(json_dump)
+    return APSchedulerJobCreate.model_validate_json(json_dump)
+
+
+def add_create_scheduled_task_to_queue(payload: APSchedulerJobCreate) -> None:
+    """Create a scheduled task service function.
+
+    We need to create a apscheduler job, and put the workflow and schedule_id in
+    the linker table workflows_apscheduler_jobs.
+
+    Args:
+        payload: APSchedulerJobCreate The scheduled task to create.
+    """
+    bytes_dump = serialize_payload(payload)
+    redis_connection.lpush(SCHEDULER_QUEUE, bytes_dump)
+    logger.info("Added create scheduled task to queue.")
+
+
+def add_update_scheduled_task_to_queue(payload: APSchedulerJobUpdate) -> None:
+    """Update a scheduled task service function.
+
+    Args:
+        payload: APSchedulerJobUpdate The scheduled task to update.
+    """
+    bytes_dump = serialize_payload(payload)
+    redis_connection.lpush(SCHEDULER_QUEUE, bytes_dump)
+    logger.info("Updated scheduled task to queue.")
+
+
+def add_delete_scheduled_task_to_queue(payload: APSchedulerJobDelete) -> None:
+    """Delete a scheduled task service function.
+
+    We need to delete a apscheduler job, and remove the workflow and schedule_id in
+    the linker table workflows_apscheduler_jobs.
+
+    Args:
+        payload: APSchedulerJobDelete The scheduled task to delete.
+    """
+    bytes_dump = serialize_payload(payload)
+    redis_connection.lpush(SCHEDULER_QUEUE, bytes_dump)
+    logger.info("Added delete scheduled task to queue.")
+
+
+def get_linker_entries_by_schedule_id(schedule_id: str) -> list[WorkflowApschedulerJob]:
+    """Get linker table entries by schedule ID.
+
+    Args:
+        schedule_id: str The schedule ID.
+
+    Returns:
+        list[WorkflowApschedulerJob]: The linker table entries.
+    """
+    return db.session.query(WorkflowApschedulerJob).filter(WorkflowApschedulerJob.schedule_id == schedule_id).all()
+
+
+def _add_linker_entry(workflow_id: UUID, schedule_id: str) -> None:
+    """Add an entry to the linker table workflows_apscheduler_jobs.
+
+    Args:
+        workflow_id: UUID The workflow ID.
+        schedule_id: str The schedule ID.
+    """
+    with db.session.begin():
```

Comment:
Explicitly creating a session is not needed, the Database class takes care of that :)